nemo/collections/asr/parts/mixins/ipl_mixin.py

```diff
@@ -0,0 +1,529 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import os
+import tempfile
+from typing import List
+import editdistance
+import torch
+from tqdm.auto import tqdm
+
+from nemo.collections.asr.data.audio_to_text import expand_sharded_filepaths
+from nemo.collections.asr.parts.utils.ipl_utils import *
+
+
+class IPLMixin:
+    """
+    Adds ability to do iterative pseudo-labeling.
+    To use ipl parameters should be given in the config as well as max_steps should be proovided in trainer.
+
+    ipl:
+        m_epochs: 0
+        restore_pc: false
+        manifest_filepath: /path/to/manifest
+        tarred_audio_filepaths:  /path/to/tarred/
+        is_tarred: false
+        dataset_weights: 1
+        limit_train_batches: 170
+        cache_manifest: /path/for/cache/manifest
+        dropout: 0.1
+        n_l_epochs: 0
+        p_cache: 0.2
+        cache_prefix: prefix_for_cache - (optional for non-tar datasets)
+        batch_size: 128
+
+    Call
+
+        * ``self.setup_ipl()``
+          in the init method
+        * ``self.maybe_do_ipl()``
+          in the `on_train_epoch_end` method.
+    """
+
+    def setup_ipl(self):
```

Comment:
make sure to add doc strings for all functions and classes

Reply:
Done